contribs/gnogenesis/internal/txs/txs_add_packages.go

```diff
@@ -117,3 +131,55 @@ func execTxsAddPackages(
 
 	return nil
 }
+
+func signTxs(txs []gnoland.TxWithMetadata, signer *gnoclient.SignerFromKeybase) error {
+	for index, tx := range txs {
+		signBytes, err := tx.Tx.GetSignBytes(signer.ChainID, 0, 0)
+		if err != nil {
+			return fmt.Errorf("unable to load txs from directory, %w", err)
+		}
+		signature, publicKey, err := signer.Keybase.Sign(signer.Account, signer.Password, signBytes)
+		txs[index].Tx.Signatures = []std.Signature{
+			{
+				PubKey:    publicKey,
+				Signature: signature,
+			},
+		}
+		if err != nil {
+			return fmt.Errorf("unable sign tx %w", err)
+		}
+	}
+
+	return nil
+}
+
+func signerWithConfig(cfg *addPkgCfg, io commands.IO, chainID string) (*gnoclient.SignerFromKeybase, error) {
+	var (
+		keyname = integration.DefaultAccount_Name
+		pass    string
+		kb      keys.Keybase
+		err     error
+	)
+
+	if cfg.keyName != "" {
+		keyname = cfg.keyName
+		kb, err = keys.NewKeyBaseFromDir(cfg.gnoHome)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load keybase: %w", err)
+		}
+		pass, err = io.GetPassword("Enter password.", cfg.insecurePasswordStdin)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read password: %w", err)
+		}
+	} else {
+		kb = keys.NewInMemory()
+		kb.CreateAccount(integration.DefaultAccount_Name, integration.DefaultAccount_Seed, "", "", 0, 0)
+	}
+
+	return &gnoclient.SignerFromKeybase{
```

Comment:
I'm not a huge fan of using the Signer from `gnoclient`, mostly because it does a lot of redundant checks you don't really need for this scenario (check the implementation) -- notice that you never even use this signer, but its `Keybase` object.

Drop it entirely, all you need is the Keybase anyways :)

Reply:
Yes you're right I was just using it as a way to keep all the info in one struct but I did not wanted to create another struct, now I'm doing all on the main function to get the right Keybase account name etc.

let me know if you prefer another approach, thanks for the comment btw :)
09dce4322f68e8f7aa366c3c33da287d5761d8ae